packages/connector-oauth2/src/index.ts

```diff
@@ -0,0 +1,116 @@
+import type {
+  GetAuthorizationUri,
+  GetUserInfo,
+  SocialConnector,
+  CreateConnector,
+  GetConnectorConfig,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  validateConfig,
+  ConnectorType,
+  parseJson,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import got, { HTTPError } from 'got';
+import omit from 'lodash.omit';
+import snakecaseKeys from 'snakecase-keys';
+
+import { defaultMetadata, defaultTimeout, oauthConfigGlobalKeys } from './constant';
+import type { OauthConfig } from './types';
+import { oauthConfigGuard, OauthGrantType } from './types';
+import {
+  userProfileMapping,
+  getAuthorizationCodeFlowAccessToken,
+  getImplicitFlowAccessToken,
+} from './utils';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri }, setSession) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig<OauthConfig>(config, oauthConfigGuard);
+    const parsedConfig = oauthConfigGuard.parse(config);
+
+    const parameterObject = snakecaseKeys(
+      omit(parsedConfig, ...oauthConfigGlobalKeys, 'clientSecret', 'grantType')
+    );
+
+    assert(
+      setSession,
```

Comment:
Can we make "setSession" required in typing, and no need to check in the runtime?

Reply:
Good point, add LOG-4913 to track this. Let's address this in next cooldown cycle.